app/src/main/java/com/android/streetworkapp/ui/park/AddEvent.kt

```diff
@@ -0,0 +1,305 @@
+              horizontalAlignment = Alignment.CenterHorizontally) {
+                Text(
+                    text = "Event Creation",
+                    fontSize = 24.sp,
+                )
+                EventTitleSelection(event)
+                EventDescriptionSelection(event)
+                TimeSelection(event)
+                Text(
+                    text = "How many participants do you want?",
+                    fontSize = 18.sp,
+                )
+                ParticipantNumberSelection(event)
+              }
+          FloatingActionButton(
+              onClick = {
+                // TODO: use one of the future viewModel to add the event to firebase
+                navigationActions.goBack()
+              },
+              modifier =
+                  Modifier.align(Alignment.BottomCenter)
+                      .padding(40.dp)
+                      .size(width = 150.dp, height = 40.dp)
+                      .testTag("addEventButton"),
+          ) {
+            Text("Add new event")
+          }
+        }
+      }
+}
+
+/**
+ * Used to change the title of the event
+ *
+ * @param event the event that will be updated
+ */
+@Composable
+fun EventTitleSelection(event: Event) {
+  var title by remember { mutableStateOf("") }
+
+  OutlinedTextField(
+      value = title,
+      onValueChange = {
+        title = it
+        event.title = title
+      },
+      label = { Text("What kind of event do you want to create?") },
+      modifier = Modifier.testTag("titleTag").fillMaxWidth(0.9f).height(64.dp))
+}
+
+/**
+ * Used to change the description of the event
+ *
+ * @param event the event that will be updated
+ */
+@Composable
+fun EventDescriptionSelection(event: Event) {
+  var description by remember { mutableStateOf("") }
+
+  OutlinedTextField(
+      value = description,
+      onValueChange = {
+        description = it
+        event.description = description
+      },
+      label = { Text("Describe your event:") },
+      modifier = Modifier.testTag("descriptionTag").fillMaxWidth(0.9f).height(128.dp))
+}
+
+/**
+ * Used to change the maximum number of participants of the event
+ *
+ * @param event the event that will be updated
+ */
+@Composable
+fun ParticipantNumberSelection(event: Event) {
+  val minParticipants = 2 // We need at least 2 users for every event
+  val maxParticipants = 10 // An event can not have more then 10 participants
+  var sliderPosition by remember { mutableFloatStateOf(minParticipants.toFloat()) }
+
+  Column(
+      verticalArrangement = Arrangement.Center,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = Modifier.fillMaxWidth()) {
+        Slider(
+            modifier = Modifier.fillMaxWidth(0.8f).testTag("sliderMaxParticipants"),
+            value = sliderPosition,
+            onValueChange = {
+              sliderPosition = it
+              event.maxParticipants = sliderPosition.toInt()
```

Comment:
Suggestion: (this **might** be a non issue but it will depend on our later implementation.) In this composable and other composables you immediately change the value of the event on each input. However data classes are passed by reference. If someone goes in, makes changes, doesn't want to save and goes back, it could lead to the object he has locally not matching the one in the database if we don't resync. (Again this will depend on our later implementations but I think it can be dangerous)

Reply:
I understand your concern. I think that, in the context of this screen, this is not a problem since the objective is only  create locally a new Event, that will later be sent to the database. This is like a "builder" for events, that once this event is ready locally, is sent to the database. This means that there should not be a situation where an event is fetch from the database (in this screen at least), since it is only used to add new events (this would be expected probably on an "edit event" screen perhaps). Therefor, I think that having an event as an argument for every composable, that is always an event created locally to be edited by the user through the UI and then sent to the database once it is ready, is the best way to have small and understandable composable. In this context, the "pass by reference" is intended.

This is why I think that we should keep this implementation. Do you agree?

Reply:
Yeah all good, I mixed things up and thought it would also do edits for some reason. my bad.